File: Lab_06/Lab06(C++)/Exception.cpp
#include "Exception.h"

Exception::Exception(double a, double b, double c, double d) {
    this->a = a;
    this->b = b;
    this->c = c;
    this->d = d;
}

double Exception::value() {
    try {
        if (d > 41) {
            throw "Expression`s root has negative value";
        }
    }

    catch (const char* ch) {
        MessageBoxA(GetActiveWindow(), ch, "Error!", MB_ICONERROR);
        //exit(1);
    }

    try {
        if (sqrt(41 - d) - b * a + c == 0) {
            throw "Expression`s denominator is zero";
        }
    }

    catch (const char* ch) {
        MessageBoxA(GetActiveWindow(), ch, "Error message", MB_ICONERROR);
        //exit(1);
    }

    return (a * b / 4.0 - 1) / (sqrt(41 - d) - b * a + c);
}

File: Lab_06/Lab06(Csh)/Lab06(Csh)/Exception.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static System.Math;

namespace Lab06_Csh_
{
    class Exception
    {
        private double a, b, c, d;
        public Exception(double a, double b, double c, double d)
        {
            this.a = a;
            this.b = b;
            this.c = c;
            this.d = d;
        }

        public double value()
        {
            if(d > 41)
            {
                throw new DivideByZeroException("Expression`s root has negative value");
            }
            else
            {
                if(Sqrt(41 - d) - b * a + c == 0)
                {
                    throw new DivideByZeroException("Expression`s denominator is zero");
                }
                else
                {
                    return (a * b / 4.0 - 1) / (Sqrt(41 - d) - b * a + c);
                }
            }

        }
    }
}
